src/server/routes.js

```diff
@@ -6,40 +6,54 @@ const Member = require('./controllers/member');
 const router = new Router();
 
 router.get("/ideas", async (ctx, next) => {
-	const ideas = await Idea.showAll();
-	ctx.status = HttpStatus.OK;
-	ctx.body = ideas;
-	await next();
+	try {
+		const ideas = await Idea.showAll();
+		ctx.response.status = HttpStatus.OK;
+		ctx.response.body = ideas;
+	} catch (err) {
+		ctx.throw(HttpStatus.INTERNAL_SERVER_ERROR, err.message);
+	}
 });
 
 router.get("/members", async (ctx, next) => {
-	const ideas = await Member.showAll();
-	ctx.status = HttpStatus.OK;
-	ctx.body = ideas;
-	await next();
+	try {
+		const ideas = await Member.showAll();
+		ctx.response.status = HttpStatus.OK;
+		ctx.body = ideas;
+	} catch (err) {
+		ctx.throw(HttpStatus.INTERNAL_SERVER_ERROR, JSON.stringify({message: err.message}));
+	}
 });
 
 router.post("/addNewidea", async (ctx, next) => {
-	const content = ctx.request.body;
-	const idea = await Idea.create(content);
-
-	ctx.status = HttpStatus.OK;
-	ctx.body = idea.toJSON();
-	await next();
+	try {
+		const content = ctx.request.body;
+		const idea = await Idea.create(content);
+		ctx.status = HttpStatus.OK;
+		ctx.body = idea.toJSON();
+	} catch (err) {
+		ctx.throw(HttpStatus.INTERNAL_SERVER_ERROR, err.message);
+	}
 });
 
 router.post('/join', async (ctx, next) => {
-	const content = ctx.request.body;
-	const idea = await Member.addNewMember(content.id, content.member);
-	ctx.status = HttpStatus.OK;
-	ctx.body = idea.toJSON();
-	await next();
+	try {
+		const content = ctx.request.body;
+		const idea = await Member.addNewMember(content.id, content.member);
+		ctx.response.status = HttpStatus.OK;
+		ctx.response.body = idea.toJSON();
+	} catch (err) {
+		ctx.throw(HttpStatus.INTERNAL_SERVER_ERROR, JSON.stringify({message: err.message}));
+	}
 });
 
 router.get('/ideas/delete', async (ctx, next) => {
```

Comment:
The routes again ;)

This one should be `router.delete('/ideas')`. This means: The HTTP verb is [`DELETE`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE), the _resource_ you want to delete is `/ideas`.

The corresponding request from the client looks like this:

`DELETE /ideas`

So, in the end the schema boils down to:

`HTTP verb /resourceClass/individualID`

Examples:

* Retrieve all ideas: `GET /ideas`
* Retrieve a specific idea with id `1`: `GET /ideas/1`
* Delete idea with id `1`: `DELETE /ideas/1`
* Create an idea: `POST /ideas` _(with idea content in `body` as `JSON` as you already do)_
* Update idea with id `1`: `PUT /ideas/1` with updated idea (the whole (!) new object) in `body` as `JSON`
* Delete all ideas: `DELETE /ideas`